### PR TITLE
📑 570 adding custom tab for database and messaging span types

### PR DESCRIPTION
**Pull request by xoscar**
This PR updates the set of tabs that we display for the span details when displaying the information for database, messaging, and HTTP types.

## Changes

- New empty attribute list component.
- New way of filtering the span attributes.
- Including the OTEL supported package for semantic definitions.

## Fixes

- https://github.com/kubeshop/tracetest/issues/570
- https://github.com/kubeshop/tracetest/issues/569

## Checklist

- [x] tested locally
- [x] added new dependencies
- [ ] updated the docs
- [x] added a test

https://www.loom.com/share/b695b000bd584ee0808264fd9a6bc6f2

### CODE REVIEW

File: web/src/components/AttributeList/AttributeList.tsx [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
 const AttributeList: React.FC<IAttributeListProps> = ({attributeList, onCreateAssertion}) => {
-  return (
-    <S.AttributeList>
+  return attributeList.length ? (
+    <S.AttributeList data-cy="attribute-list">
       {attributeList.map(attribute => (
         <AttributeRow attribute={attribute} key={attribute.key} onCreateAssertion={onCreateAssertion} />
       ))}
     </S.AttributeList>
+  ) : (
```
Comment (xoscar) [inline web/src/components/AttributeList/AttributeList.tsx:18]: Adding new empty attribute list component

File: web/src/components/SpanDetail/SpanDetail.tsx [hunk 2]
```diff
@@ -64,7 +58,7 @@ const SpanDetail: React.FC<TSpanDetailProps> = ({span}) => {
   return (
     <S.SpanDetail>
       <SpanHeader title={title} />
-      <Component span={span} onCreateAssertion={onCreateAssertion} />
+      <SpanDetailTabs onCreateAssertion={onCreateAssertion} span={span} />
```
Comment (xoscar) [inline web/src/components/SpanDetail/SpanDetail.tsx:61]: Removing the factory as its not required atm

File: web/src/constants/SemanticGroupNames.constants.ts [hunk 3]
```diff
@@ -34,129 +24,121 @@ export const SemanticGroupNamesToText = {
 
 export const SemanticGroupNameNodeMap: Record<SemanticGroupNames, {primary: string[]; type: string}> = {
   [SemanticGroupNames.Http]: {
-    primary: ['http.target', 'service.name'],
+    primary: [Attributes.HTTP_TARGET, Attributes.SERVICE_NAME],
```
Comment (xoscar) [inline web/src/constants/SemanticGroupNames.constants.ts:27]: No more magic strings, now everything comes from the OTEL semantic conventions package

File: web/src/constants/Span.constants.ts [hunk 4]
```diff
@@ -1,13 +1,88 @@
...
+  response = 'response',
+  metadata = 'metadata',
+  operation = 'operation',
+  consumer = 'consumer',
+  producer = 'producer',
+  custom = 'custom',
+  all = 'all',
+}
+
+export const SpanAttributeSections: Record<SemanticGroupNames, Record<string, TValueOfAttributes[]>> = {
```
Comment (xoscar) [inline web/src/constants/Span.constants.ts:17]: Defining the different tabs and sections for each of the semantic group names

File: web/src/services/SpanAttribute.service.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+const filterAttributeList = (attributeList: TSpanFlatAttribute[], attrKeyList: string[]): TSpanFlatAttribute[] => {
+  return attrKeyList.reduce<TSpanFlatAttribute[]>((list, key) => {
+    const foundAttr = attributeList.find(attr => attr.key === key);
+
+    return foundAttr ? list.concat(foundAttr) : list;
+  }, []);
+};
+
+const SpanAttributeService = () => ({
```
Comment (xoscar) [inline web/src/services/SpanAttribute.service.ts:16]: Filters the attribute list and returns the list of sections

File: web/src/components/AttributeList/EmptyAttributeList.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+const EmptyAttributeList: React.FC = () => {
+  return (
+    <S.EmptyAttributeList data-cy="empty-attribute-list">
+      <S.EmptyIcon />
+      <S.EmptyTitle>Looking for attributes?</S.EmptyTitle>
+      <S.EmptyText>
+        Take a look a the open telemetry specification for{' '}
+        <a href={TRACE_SEMANTIC_CONVENTIONS_URL}>trace semantic conventions</a> and here for{' '}
+        <a href={RESOURCE_SEMANTIC_CONVENTIONS_URL}>resource semantic conventions</a>.
```
Comment (jorgeepc) [inline web/src/components/AttributeList/EmptyAttributeList.tsx:12]: maybe a `target="_blank"` here?

File: web/src/components/SpanDetail/SpanDetail.tsx [hunk 7]
```diff
@@ -1,19 +1,17 @@
...
+import {TSpan, TSpanFlatAttribute} from '../../types/Span.types';
 import SpanHeader from './SpanHeader';
 import * as S from './SpanDetail.styled';
+import {useAssertionForm} from '../AssertionForm/AssertionFormProvider';
+import {CompareOperator} from '../../constants/Operator.constants';
+import OperatorService from '../../services/Operator.service';
+import SpanDetailTabs from './SpanDetailTabs';
+import {useAppSelector} from '../../redux/hooks';
+import TestDefinitionSelectors from '../../selectors/TestDefinition.selectors';
+import {TResultAssertions} from '../../types/Assertion.types';
```
Comment (jorgeepc) [inline web/src/components/SpanDetail/SpanDetail.tsx:14]: Are you reverting back to relative paths? 😄

File: web/src/components/SpanDetail/__tests__/__snapshots__/SpanDetail.test.tsx.snap [hunk 8]
```diff
@@ -0,0 +1,187 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (jorgeepc) [inline web/src/components/SpanDetail/__tests__/__snapshots__/SpanDetail.test.tsx.snap:1]: Not for now, but we need to start removing these snapshots.

**Review by jorgeepc — APPROVED** [review]
Great work my friend! 🔥

File: web/src/components/AttributeList/EmptyAttributeList.tsx (hunk 6 above)
Comment (xoscar) [reply web/src/components/AttributeList/EmptyAttributeList.tsx:12]: Done

**Pull request merged**